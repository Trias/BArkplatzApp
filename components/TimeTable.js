import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';
import { Table, Row, Rows } from 'react-native-table-component';

export default class TimeTable extends Component {
  constructor(props) {
    super(props);
    this.state = {
      tableHead: [props.name],
      tableData: [
        ['1. - 3. Stunde', 'je 1.50€' ],
        ['4. Stunde', '1.00€'],
        ['Tageshöchstgebühr', '5.50€'],
        ['Kongressticket 7-Tage', '25.00€'],
        ['Kongressticket 28-Tage', '50.00€'],
      ]
    }
  }

  render() {
    const state = this.state;
    return (
      <View style={styles.container}>
        <Table borderStyle={{borderWidth: 1, borderColor: '#000'}}>
          <Row data={state.tableHead} style={styles.head} textStyle={styles.texthead}/>
          <Rows data={state.tableData} textStyle={styles.text}/>
        </Table>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: { 
    flex: 1, 
    width: "100%",
    padding: 16, 
    backgroundColor: '#fff' },
  head: { 
    height: 40, 
    backgroundColor: '#BB2D37', 
    alignItems: "center"},
  text: { 
  margin: 6 },
  texthead:{
    fontSize: 15,
    marginTop: 10, 
    padding: 5,
     color: '#fff',
    alignContent: "center",
    textAlign: "center"
  },
});

