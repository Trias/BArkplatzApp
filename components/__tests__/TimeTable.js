import React from 'react';
import renderer from 'react-test-renderer';

import TimeTable from '../TimeTable';

it(`renders correctly`, () => {
  const tree = renderer.create(<TimeTable
  />).toJSON();

  expect(tree).toMatchSnapshot();
});
