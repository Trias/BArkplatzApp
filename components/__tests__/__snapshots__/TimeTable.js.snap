// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "padding": 16,
      "width": "100%",
    }
  }
>
  <View
    style={
      Array [
        undefined,
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#000",
          "borderLeftWidth": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
          0,
          Object {
            "flexDirection": "row",
            "overflow": "hidden",
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#BB2D37",
            "height": 40,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderColor": "#000",
              "borderRightWidth": 1,
              "borderTopWidth": 1,
            },
            Object {
              "justifyContent": "center",
            },
            undefined,
            undefined,
            undefined,
            Object {
              "flex": 1,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "alignContent": "center",
                "color": "#fff",
                "fontSize": 15,
                "marginTop": 10,
                "padding": 5,
                "textAlign": "center",
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Array [
          0,
          0,
        ]
      }
    >
      <View
        style={
          Array [
            undefined,
            0,
            Object {
              "flexDirection": "row",
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#000",
                "borderRightWidth": 1,
                "borderTopWidth": 1,
              },
              Object {
                "justifyContent": "center",
              },
              undefined,
              undefined,
              undefined,
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            1. - 3. Stunde
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "borderColor": "#000",
                "borderRightWidth": 1,
                "borderTopWidth": 1,
              },
              Object {
                "justifyContent": "center",
              },
              undefined,
              undefined,
              undefined,
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            je 1.50€
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
            0,
            Object {
              "flexDirection": "row",
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#000",
                "borderRightWidth": 1,
                "borderTopWidth": 1,
              },
              Object {
                "justifyContent": "center",
              },
              undefined,
              undefined,
              undefined,
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            4. Stunde
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "borderColor": "#000",
                "borderRightWidth": 1,
                "borderTopWidth": 1,
              },
              Object {
                "justifyContent": "center",
              },
              undefined,
              undefined,
              undefined,
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            1.00€
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
            0,
            Object {
              "flexDirection": "row",
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#000",
                "borderRightWidth": 1,
                "borderTopWidth": 1,
              },
              Object {
                "justifyContent": "center",
              },
              undefined,
              undefined,
              undefined,
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            Tageshöchstgebühr
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "borderColor": "#000",
                "borderRightWidth": 1,
                "borderTopWidth": 1,
              },
              Object {
                "justifyContent": "center",
              },
              undefined,
              undefined,
              undefined,
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            5.50€
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
            0,
            Object {
              "flexDirection": "row",
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#000",
                "borderRightWidth": 1,
                "borderTopWidth": 1,
              },
              Object {
                "justifyContent": "center",
              },
              undefined,
              undefined,
              undefined,
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            Kongressticket 7-Tage
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "borderColor": "#000",
                "borderRightWidth": 1,
                "borderTopWidth": 1,
              },
              Object {
                "justifyContent": "center",
              },
              undefined,
              undefined,
              undefined,
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            25.00€
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
            0,
            Object {
              "flexDirection": "row",
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "borderColor": "#000",
                "borderRightWidth": 1,
                "borderTopWidth": 1,
              },
              Object {
                "justifyContent": "center",
              },
              undefined,
              undefined,
              undefined,
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            Kongressticket 28-Tage
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "borderColor": "#000",
                "borderRightWidth": 1,
                "borderTopWidth": 1,
              },
              Object {
                "justifyContent": "center",
              },
              undefined,
              undefined,
              undefined,
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "margin": 6,
                },
                Object {
                  "backgroundColor": "transparent",
                },
              ]
            }
          >
            50.00€
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
